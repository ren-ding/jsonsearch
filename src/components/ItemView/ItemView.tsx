import React, { Component } from 'react';

export interface ItemViewProps {
    itemObject: Object
}

export default class ItemView extends Component<ItemViewProps> {
  renderObject = () =>
        Object.entries(this.props.itemObject).map(([key, value]) => {
			return (
                <tr key={key}>
                <td>{key}</td>
                <td>{value.toString()}</td>
                </tr>
			);
		});
  
  render() {
    return (
      <div>
          <table className="ui compact table">
            <thead>
              <tr>
                <th>Field</th>
                <th>Value</th>
              </tr>
            </thead>
            <tbody>
              {this.renderObject()}
            </tbody>
          </table>
        </div>
    );
  }
}