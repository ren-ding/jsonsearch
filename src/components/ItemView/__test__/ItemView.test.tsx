import React from 'react';
import ItemView, {ItemViewProps} from '../ItemView';
import renderer from 'react-test-renderer';

describe('ItemView', () => {
    let props: ItemViewProps;
    beforeEach(()=> {
        props = {
            itemObject: {variableName: 'variableValue'}
        }
    });

    it('should render to match snapshot', () => {
        const tree = renderer.create(<ItemView {...props}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});