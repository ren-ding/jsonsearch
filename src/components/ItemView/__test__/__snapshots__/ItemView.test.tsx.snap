// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemView should render to match snapshot 1`] = `
<div>
  <table
    className="ui compact table"
  >
    <thead>
      <tr>
        <th>
          Field
        </th>
        <th>
          Value
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          variableName
        </td>
        <td>
          variableValue
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
